src/control/tools/EditSelection.cpp

```diff
@@ -63,7 +66,8 @@ EditSelection::EditSelection(UndoRedoHandler* undo, Element* e, XojPageView* vie
 }
 
 EditSelection::EditSelection(UndoRedoHandler* undo, const vector<Element*>& elements, XojPageView* view,
-                             const PageRef& page) {
+                             const PageRef& page):
+        SnapToGridInputHandler(view->getXournal()->getControl()->getSettings()) {
```

Comment:
Use composition over inheritance here, since this is a "has-an" not an "is-an" relationship.